rpcs3/Emu/Cell/PPUTranslator.cpp

```diff
@@ -222,6 +222,24 @@ Function* PPUTranslator::Translate(const ppu_function& info)
 	return m_function;
 }
 
+
+// Trunc denormal/subnormal values (-FLT_MIN, FLT_MIN) to 0
+// VREFP always generates infinity giving a denormal value,
+// that causes subsequent VMADDFP/VNMSUBFP generates NaN instead of correct value.
+//
+Value* PPUTranslator::VecHandleSubnormal(Value* val)
+{
+	const auto zero_vec4 = ConstantVector::getSplat(4, ConstantInt::get(GetType<u32>(), 0u));
+	const auto subnormal_mask = ConstantVector::getSplat(4, ConstantInt::get(GetType<u32>(), 0x7f800000u));
+
+	auto val_u32  = m_ir->CreateBitCast(val, GetType<u32[4]>());	
+	auto sub_mask = m_ir->CreateAnd(subnormal_mask, val_u32);	
+	sub_mask = Solid(m_ir->CreateICmpUGT(sub_mask, zero_vec4));
+	sub_mask = m_ir->CreateBitCast(sub_mask, GetType<u32[4]>());
+	val = m_ir->CreateAnd(sub_mask, val_u32);
+	return m_ir->CreateBitCast(val, GetType<f32[4]>());
+}
+
```

Comment:
Won't this cause -0 to become +0?

Reply:
Yes, the -0 (0x8000000) will be truncate to +0(0x0), but I haven't found any side effect for the functionality yet.

Reply:
I don't think it's correct to throw away one piece of accuracy for another, but at the very least you need to document this kind of thing in the comments.

Reply:
Okay, I'll left a comment here.